core/os/android/adb/perfetto.go

```diff
@@ -29,9 +33,25 @@ import (
 	"github.com/google/gapid/core/log"
 )
 
+// DeleteFile deletes a file on the device.
+func (b *binding) DeleteFile(ctx context.Context, traceFile string) error {
+	_, err := b.Shell("rm", "-f", traceFile).Call(ctx)
+	if err != nil {
+		err = errors.New(fmt.Sprintf("Failed to delete %s. Try root", traceFile))
+	}
+	return err
+}
+
 // StartPerfettoTrace starts a perfetto trace on this device.
 func (b *binding) StartPerfettoTrace(ctx context.Context, config *perfetto_pb.TraceConfig, out string, stop task.Signal) error {
+
+	// Silently attempt to delete the old trace in case the user has changed
+	// If the user has changed from shell => root, this is fine. For root => shell,
+	// the root user will have still have to manually delete the old trace if it exists
+	b.DeleteFile(ctx, out)
```

Comment:
```
if err := deleteFile(b, ctx, out); err != nil {
  return err
}
```

Reply:
We don't want to return an error for the reasons mentioned in the description.